atlasdb-backup/src/test/java/com/palantir/atlasdb/backup/transaction/Transactions3TableInteractionTest.java

```diff
@@ -0,0 +1,178 @@
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.codec.binary.Hex;
+import org.junit.Before;
+import org.junit.Test;
+
+public class Transactions3TableInteractionTest {
+    private static final FullyBoundedTimestampRange RANGE = FullyBoundedTimestampRange.of(Range.closed(5L, 500L));
+    private static final String KEYSPACE = "keyspace";
+
+    private final RetryPolicy mockPolicy = mock(RetryPolicy.class);
+    private final TransactionsTableInteraction interaction = new Transactions3TableInteraction(RANGE, mockPolicy);
+    private final TableMetadata tableMetadata = mock(TableMetadata.class, RETURNS_DEEP_STUBS);
+
+    @Before
+    public void setupMock() {
+        when(tableMetadata.getKeyspace().getName()).thenReturn(KEYSPACE);
+        when(tableMetadata.getName()).thenReturn(TransactionConstants.TRANSACTIONS2_TABLE.getTableName());
+    }
+
+    @Test
+    public void extractCommittedTimestampTest() {
+        TransactionTableEntry entry = interaction.extractTimestamps(createRow(150L, 200L));
+        assertThat(entry.getStartTimestamp()).isEqualTo(150L);
+        assertThat(entry.getCommitTimestamp()).hasValue(200L);
+    }
+
+    @Test
+    public void extractStagingCommitTimestampTest() {
+        TransactionTableEntry entry =
+                interaction.extractTimestamps(createRow(150L, PutUnlessExistsValue.staging(200L)));
+        assertThat(entry.getStartTimestamp()).isEqualTo(150L);
+        assertThat(entry.getCommitTimestamp()).hasValue(200L);
+    }
+
+    @Test
+    public void extractAbortedTimestampTest() {
+        TransactionTableEntry entry = interaction.extractTimestamps(createAbortedRow(150L));
+        assertThat(entry.getStartTimestamp()).isEqualTo(150L);
+        assertThat(entry.getCommitTimestamp()).isEmpty();
+    }
+
+    @Test
+    public void extractStagingAbortedTimestampTest() {
+        TransactionTableEntry entry = interaction.extractTimestamps(
+                createRow(150L, PutUnlessExistsValue.staging(TransactionConstants.FAILED_COMMIT_TS)));
+        assertThat(entry.getStartTimestamp()).isEqualTo(150L);
+        assertThat(entry.getCommitTimestamp()).isEmpty();
+    }
+
+    @Test
+    public void isRowAbortedTest() {
+        assertThat(interaction.isRowAbortedTransaction(createRow(12345L, 234234L)))
+                .isFalse();
+        assertThat(interaction.isRowAbortedTransaction(createRow(235L, PutUnlessExistsValue.staging(2345L))))
+                .isFalse();
+        assertThat(interaction.isRowAbortedTransaction(createAbortedRow(234L))).isTrue();
+        assertThat(interaction.isRowAbortedTransaction(
+                        createRow(23546L, PutUnlessExistsValue.staging(TransactionConstants.FAILED_COMMIT_TS))))
+                .isTrue();
+    }
+
+    @Test
+    public void getAllRowsInPartition() {
+        Range<Long> rangeWithinOnePartition = Range.closed(100L, 1000L);
+        Transactions2TableInteraction txnInteraction =
```

Comment:
Oops!

Reply:
Also, please edit the following few tests to use Transactions3.

The (mostly) duplicated form here is not ideal. I think we could improve upon this maybe with some kind of parameterised testing? Though if it's too much effort I would be ok with this.

Separately: Are we blocking the rollout of Transactions3 behind the AtlasDB backup service work? (I don't think we should.)